pkg/provider/core.go

```diff
@@ -430,6 +441,29 @@ func FindInstanceByFolderAndName(ctx context.Context, sdk *ycsdk.SDK, folderID s
 	return result.Instances[0], nil
 }
 
+func findUnusedAddressFromListByFolder(ctx context.Context, sdk *ycsdk.SDK, folderID string, addresses []string) (string, error) {
+	result, err := sdk.VPC().Address().List(ctx, &vpc.ListAddressesRequest{
+		FolderId: folderID,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	for _, item := range result.Addresses {
+		if !item.GetUsed() {
+			if external := item.GetExternalIpv4Address().GetAddress(); external != "" {
+				for _, address := range addresses {
+					if external == address {
+						return external, nil
+					}
+				}
+			}
+		}
+	}
+
```

Comment:
Though this addresses, previous comment from @himanshu-kun, I think this new implementation introduces higher depth nesting of `for` and `if` blocks which may be a good thing from readability perspective. May be here's an alternative suggestion (please indent it accordingly). 

```suggestion
for _, item := range result.Addresses{
	if !item.GetUsed(){
		external := item.GetExternalIpv4Address().GetAddress()
		if external == "" {
			continue
		}
		for _, address := range addresses {
			if external == address {
				return external, nil
			}
		}
	}
}
```

Reply:
Thanks for explanation. Changed like this.